### PR TITLE
Transducers draft post

### PR DESCRIPTION
**Pull request by davidnormo**
Let me know if I need to spend more time clarifying/simplifying any of the points. Does it all make sense?

### CODE REVIEW

**Comment by iplog** [comment]
Will review later today!

**Comment by davidnormo** [comment]
Gemma Hitchens has had a look and make some small grammatical tweaks.

File: _drafts/2016-10-24-transducers.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+It's also a nicer abstraction than working with reducing functions directly. Although
+there is a complexity overhead, transducers are still actually quite readable.
+
+And because they are all just functions with the same signature you can get clever and compose pipelines!
+
+<figure><figcaption>Fig. 3</figcaption></figure>
+
+```javascript
+const xf1 = comp( map(...), pick(...) );
+const xf2 = comp( keep(...), filter(...) );
```
Comment (iplog) [inline _drafts/2016-10-24-transducers.md]: I think I would have a short paragraph about the order in the composition. I guess that's not the same to do `comp( keep(...), filter(...) );` and `comp( filter(...), keep(...) );` right?

File: _drafts/2016-10-24-transducers.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+const xf = comp (
+  pick(3), // `pick` returns a transducer that lets n elements pass
+  map(x => x + 1)
+);
+transduce(xf, sum, 0, [1,2,3,4,5]); // 9
+```
+
+Pretty neat huh? Composition is less restrictive than method chaining as traditionally
+provided by lodash/underscore <sup><a href="#ref1">[4]</a></sup>. (Note: Lodash now supports
+functional composition via `flow`.)
```
Comment (iplog) [inline _drafts/2016-10-24-transducers.md]: and the new lodash FP functions. I would add a link to their guide https://github.com/lodash/lodash/wiki/FP-Guide

File: _drafts/2016-10-24-transducers.md [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+// our `map` in terms of reduce
+const result = tmp.reduce((acc, x) => {
+  acc += x + 1;
+  return acc;
+}, 0 );
+
+result; // 9
+```
+
+You'll notice that `tmp` is just the result of our `pick` reduce. This is an
```
Comment (iplog) [inline _drafts/2016-10-24-transducers.md]: You could point that you can achieve the same with lodash FP and composition

File: _drafts/2016-10-24-transducers.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+There are many more questions/answers for using other solutions which aim to solve
+the same problems that transducers aim to solve. So if you get stuck with transducers,
+there is a greater risk that you will be stuck for longer.
+
+3. **You might not get all the benefits of transducers**<br />
+Unless you have ONE of the following, I wouldn't suggest to use transducers in a Javscript
+project:
+
+- Your reducing function logic is becoming unwieldy and you'd benefit from a more composable approach OR
+- You're suffer from performance problems because of reasons stated in <br /><a href="#performant">section 3</a> OR
```
Comment (iplog) [inline _drafts/2016-10-24-transducers.md]: broken link on GH. expecting `3-performant`

File: _drafts/2016-10-24-transducers.md [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+> ...transducers create no intermediate collections. If you want to apply several transformations, usually each one is performed in order, creating a new collection each time.
+>
+> Transducers, however, take one item off the collection at a time and fire it through the whole transformation pipeline. So it doesn't need any intermediate collections; each value runs through the pipeline separately.
+
+If we break the rules of big O notation for a minute, fig. 1 is roughly O(n) and
+fig. 5 is O(2n). Strictly speaking we drop constants and low-order terms in big O notation so they are
+both O(n) but benchmarks point out that one is more performant than the other<sup><a href="#ref8234">[5]</a></sup>.
+
+## What are the cons?
```
Comment (iplog) [inline _drafts/2016-10-24-transducers.md]: Would it be worth to add something about the way you can break it if you do a bad closure?

File: _drafts/2016-10-24-transducers.md [hunk 6]
```diff
@@ -0,0 +1,219 @@
+---
+layout: post
+title:  "Reuse logic, avoid duplication, maintain performance"
+date:   2016-10-21 11:17:48 +0000
+author: davidnormo
+---
+
+... with transducers!
```
Comment (davidnormo) [inline _drafts/2016-10-24-transducers.md]: I didn't think of it that way. I just thought it was a continuation from the title. But maybe the title should read:

"Reuse logic, avoid duplication, maintain performance with transducers"

What do you think?

Comment (iplog) [inline _drafts/2016-10-24-transducers.md]: interesting. I really read it as the end of the conversation you could have with someone and like the "obvious" answer. I don't mind. My comment is only valid the conversation effect was the intent. I think I would avoid splitting the title then.

File: _drafts/2016-10-24-transducers.md (hunk 5 above)
Comment (davidnormo) [inline _drafts/2016-10-24-transducers.md]: I don't think you can break transducers with messy closures. They can run slowly, but if you provide the correct functions you won't break it.

**Pull request merged**